### PR TITLE
New integration

### PR DESCRIPTION
**Pull request by phucla0122**
- Replace open3d with scikit-learn library -> Overall scikit is better due to it is capable of handling larger datasets along with having more algorithms that support cluster detection
- Create a new lidar object recognition package, which implements the scikit-learn library
- Remove ROS Navigation package completely from the system

### CODE REVIEW

**Review by nicolastuttle — CHANGES_REQUESTED** [review]
Could you make removing ROS Nav its own PR? It's a pretty big set of changes on its own and I don't think it should be affected by your new lidar module.

**Comment by robellgabriel** [comment]
> Could you make removing ROS Nav its own PR? It's a pretty big set of changes on its own and I don't think it should be affected by your new lidar module.

I agree, splitting them into seperate PRs make it a lot easier to review your changes

**Comment by phucla0122** [comment]
> > Could you make removing ROS Nav its own PR? It's a pretty big set of changes on its own and I don't think it should be affected by your new lidar module.
> 
> I agree, splitting them into seperate PRs make it a lot easier to review your changes

Alright, I'll do that then

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,6 @@ catkin_ws/*.pyc
 src/cruise_control/scripts/speed_output.txt
 .idea
 
+src/ros_navigation/src/binvox maps/viewvox.exe
```
Comment (nicolastuttle) [inline .gitignore]: Are these files still used? Looks like old ROS nav stuff

File: src/lidar_object_recognition_2/scripts/DBSCAN_variable.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+20
```
Comment (nicolastuttle) [inline src/lidar_object_recognition_2/scripts/DBSCAN_variable.txt:1]: What are these numbers for?

File: src/sensors/scripts/sensor.launch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<launch>
```
Comment (nicolastuttle) [inline src/sensors/scripts/sensor.launch]: More ROS Nav stuff that can be removed I think

**Review by nicolastuttle — CHANGES_REQUESTED** [review]
Couple questions about the new code

File: src/lidar_object_recognition_2/scripts/DBSCAN_variable.txt (hunk 2 above)
Comment (phucla0122) [reply src/lidar_object_recognition_2/scripts/DBSCAN_variable.txt:1]: Those are similar to the old library work, those number represent the minimum samples and epsilon for cluster detection

File: .gitignore (hunk 1 above)
Comment (phucla0122) [reply .gitignore]: Yea, that's not suppose to be there

File: .gitignore [hunk 4]
```diff
@@ -20,3 +20,6 @@ catkin_ws/*.pyc
...
+src/ros_navigation/src/binvox maps/binvox.exe
+src/ros_navigation/src/binvox maps/glut32.dll
```
Comment (robellgabriel) [inline .gitignore]: these aren't necessary as the ROS Nav package was deleted

File: src/sensors/scripts/sensor.launch (hunk 3 above)
Comment (phucla0122) [reply src/sensors/scripts/sensor.launch]: Yes

File: .gitignore (hunk 4 above)
Comment (robellgabriel) [reply .gitignore]: nevermind, I see nick already mentioned this

File: src/lidar_object_recognition_2/scripts/DBSCAN_variable.txt (hunk 2 above)
Comment (nicolastuttle) [reply src/lidar_object_recognition_2/scripts/DBSCAN_variable.txt:1]: Ok. We might want to move this into a config class with other constants in the future.

**Review by robellgabriel — APPROVED** [review]
Nice find! this new object detection looks promising

**Review by nicolastuttle — APPROVED** [review]

**Pull request merged**